minidump/src/minidump.rs

```diff
@@ -1397,19 +1384,23 @@ impl MinidumpModuleList {
     pub fn new() -> MinidumpModuleList {
         MinidumpModuleList {
             modules: vec![],
-            modules_by_addr: RangeMap::new(),
+            sorted_modules: vec![],
         }
     }
     /// Create a `MinidumpModuleList` from a list of `MinidumpModule`s.
     pub fn from_modules(modules: Vec<MinidumpModule>) -> MinidumpModuleList {
-        let modules_by_addr = modules
-            .iter()
-            .enumerate()
-            .map(|(i, module)| (module.memory_range(), i))
-            .into_rangemap_safe();
+        let mut sorted_modules: Vec<usize> = (0..modules.len()).collect();
+        sorted_modules.sort_by_key(|i| modules[*i].base_address());
+        sorted_modules.dedup_by(|ni, ci| {
+            let cur = &modules[*ci];
+            let next = &modules[*ni];
+            let cur_end_address = cur.base_address().saturating_add(cur.size());
+            let next_end_address = next.base_address().saturating_add(next.size());
```

Comment:
More saturating that should probably be checked. This one's harder to remove though, so maybe it's fine?